src/mlpack/methods/ann/layer/nearest_interpolation.hpp

```diff
@@ -0,0 +1,106 @@
+//
+/**
+ * @filer methods/ann/layer/nearest_interpolation.hpp
+ * @author Andrew Furey
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_HPP
+#define MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * Definition and Implementation of the Nearest Interpolation Layer.
+ *
+ * Nearest interpolation is an mathematical technique, primarily used for
+ * scaling purposes. The input should be a 2D matrix and it can have
+ * a number of channels/units.
+ *
+ * @tparam InputType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam MatType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template<typename MatType = arma::mat>
+class NearestInterpolationType : public Layer<MatType>
+{
+ public:
+  //! Create the NearestInterpolation object.
+  NearestInterpolationType();
+
+  /**Create NearestInterpolation Object with the same scaleFactor along 
+   * each dimension
```

Comment:
@andrewfurey21 can you issue a follow-up PR that documents that `scaleFactors` must have two elements?  (And that we plan to generalize that eventually?)